web/src/containers/P2P/Offers/P2POffersModal.tsx

```diff
@@ -199,13 +199,18 @@ const P2POffersModal: FC<Props> = (props: Props): ReactElement => {
 
     const handleClickTradeAll = React.useCallback(() => {
         if (side === 'buy') {
-            setTradeAmount(topLimit.toString());
+            const maxLimit = +topLimit * +price;
+            if ((+available * +price) < maxLimit) {
+                handleReceiveChange(available.toString());
+            } else {
+                handleAmountChange(maxLimit.toString());
+            }
         } else {
             const availableBalance = wallets.find(w => w.currency === currencyCode.toLowerCase())?.balance || 0;
             setTradeAmount(availableBalance.toString());
         }
         setClickAll(true);
-    }, [side, currencyCode, wallets]);
+    }, [side, currencyCode, wallets, available, topLimit, price]);
```

Comment:
add maxLimit

Reply:
idk why we need to add maxLimit;
maxLimit is a value calculated from topLimit and price, and we already have these params here